### PR TITLE
[Gettr] Improve extractor

### PR DESCRIPTION
**Pull request by i6t**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

- Detect embedded posts/videos
- Fix timestamp scale
- Extract subtitles
- Add expect types to `try_get`

### CODE REVIEW

File: yt_dlp/extractor/gettr.py [hunk 1]
```diff
@@ -50,31 +52,52 @@ class GettrIE(GettrBaseIE):
...
-        user_data = try_get(api_data, lambda x: x['aux']['uinf'][post_data['uid']]) or {}
+        user_data = try_get(api_data, lambda x: x['aux']['uinf'][post_data['uid']], dict) or {}
 
         if post_data.get('nfound'):
             raise ExtractorError(post_data.get('txt'), expected=True)
 
+        player_type = post_data.get('p_type')
+        shared_post_id = traverse_obj(api_data, ('aux', 'shrdpst', '_id'), ('data', 'rpstIds', 0), expected_type=str)
+
+        if player_type and player_type == 'stream':
```
Comment (Ashish0804) [inline yt_dlp/extractor/gettr.py:85]: if player_type check is not needed.
`None != 'stream'`

File: yt_dlp/extractor/gettr.py [hunk 2]
```diff
@@ -50,43 +51,70 @@ class GettrIE(GettrBaseIE):
...
+                return self.url_result(embed_url)
+            elif shared_post_id:
+                return self.url_result(f'https://gettr.com/post/{shared_post_id}', ie='Gettr', video_id=shared_post_id)
+            else:
+                raise ExtractorError('There\'s no video in this post.')
 
         title = description = str_or_none(
-            post_data.get('txt') or self._og_search_description(webpage))
+            post_data.get('txt')
+            or self._og_search_description(webpage))
```
Comment (pukkandan) [inline yt_dlp/extractor/gettr.py:106]: unnecessary change

File: yt_dlp/extractor/gettr.py [hunk 3]
```diff
@@ -50,43 +51,70 @@ class GettrIE(GettrBaseIE):
...
 
         uploader = str_or_none(
             user_data.get('nickname')
-            or remove_end(self._og_search_title(webpage), ' on GETTR'))
+            or self._search_regex(r'^(.+?) on GETTR', self._og_search_title(webpage, default=''), 'uploader', fatal=False))
```
Comment (pukkandan) [inline yt_dlp/extractor/gettr.py:110]: example where old code fails?

Comment (i6t) [reply yt_dlp/extractor/gettr.py:110]: They changed the og:title to include the body of the post.
`<meta property="og:title" content="EpochTV on GETTR : Ethiopia and the ...">`

File: yt_dlp/extractor/gettr.py [hunk 4]
```diff
@@ -106,7 +106,7 @@ def _real_extract(self, url):
 
         uploader = str_or_none(
             user_data.get('nickname')
-            or self._search_regex(r'^(.+?) on GETTR', self._og_search_title(webpage, default=''), 'uploader', fatal=False))
+            or self._search_regex(r'^(.+?) on GETTR : ', self._og_search_title(webpage, default=''), 'uploader', fatal=False))
```
Comment (pukkandan) [inline yt_dlp/extractor/gettr.py:109]: no, the other one was better. I was only asking for clarification

File: yt_dlp/extractor/gettr.py [hunk 5]
```diff
@@ -50,43 +51,69 @@ class GettrIE(GettrBaseIE):
...
+            else:
+                raise ExtractorError('There\'s no video in this post.')
 
         title = description = str_or_none(
             post_data.get('txt') or self._og_search_description(webpage))
 
         uploader = str_or_none(
             user_data.get('nickname')
-            or remove_end(self._og_search_title(webpage), ' on GETTR'))
+            or self._search_regex(r'^(.+?) on GETTR : ', self._og_search_title(webpage, default=''), 'uploader', fatal=False))
```
Comment (pukkandan) [inline yt_dlp/extractor/gettr.py:109]:
```suggestion
            or self._search_regex(r'^(.+?) on GETTR', self._og_search_title(webpage, default=''), 'uploader', fatal=False))
```
